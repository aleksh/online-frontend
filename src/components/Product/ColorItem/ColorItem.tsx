import cx from "classnames";
import * as React from "react";
import VOAttribute from "../../../VO/VOAttribute";
import Styles from "./Styles.module.scss";

interface IColorItemProps {
	color?: string;
	active?: boolean;
	item?: VOAttribute;
	click?: any;
}

export interface IColorItemState {}

class ColorItem extends React.Component<IColorItemProps, IColorItemState> {
	_handleClick = (event: any) => {
		event.preventDefault();
		const { active, item, click } = this.props;

		if (!active && click) {
			click(item);
		}
	};

	public render() {
		const { item, active, color } = this.props;

		const classes = cx(
			{ [Styles.Color]: true },
			{ [Styles.Active]: active }
		);
        
		const bgColor: any = item ? item.attribute_value : color;

		return (
			<span
				className={classes}
				style={{ background: `${bgColor}` }}
				onClick={this._handleClick}
			/>
		);
	}
}

export default ColorItem;
