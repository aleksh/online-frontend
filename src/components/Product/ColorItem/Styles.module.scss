.Color {
    width:26px;
    height: 26px; 
    display: inline-block;
    cursor: pointer;
    margin: 3px;
    border-radius: 15px;
    border: 3px solid #cccccc;
}

.Active {
    border: 3px solid #f62f5e;
}