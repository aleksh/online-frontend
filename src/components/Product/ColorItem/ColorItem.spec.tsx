import { shallow } from "enzyme";
import React from "react";
import ColorItem from "./ColorItem";

describe("ColorItem", () => {
	let wrapper;
	const mockClick = jest.fn();

	beforeEach(() => {
		wrapper = shallow(
			<ColorItem
				item={{
					attribute_name: "Color",
					attribute_value_id: 6,
					attribute_value: "White"
				}}
				active={false}
				click={mockClick}
			/>
		);
	});

	describe("TaskRow rendering", () => {
		it("render correctly TaskRow component", () => {
			expect(wrapper).toMatchSnapshot();
		});

		it("should have 1 span", () => {
			expect(wrapper.find("span")).toHaveLength(1);
		});

		it("render correct Click prop type", () => {
			expect(wrapper.props().onClick).toBeInstanceOf(Function);
		});

		it("render correct className prop type", () => {
			expect(wrapper.props().className).toBe("Color");
		});

		it("render correct className prop type", () => {
			expect(wrapper.simulate("click", { preventDefault: () =>{} }));
			expect(mockClick.mock.calls.length).toEqual(1);
		});
	});
});
