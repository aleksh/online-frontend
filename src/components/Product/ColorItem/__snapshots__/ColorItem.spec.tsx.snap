// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorItem TaskRow rendering render correctly TaskRow component 1`] = `ShallowWrapper {}`;
